skyportal/utils/catalog.py

```diff
@@ -370,3 +372,143 @@ def query_kowalski(
+    """
+
+    time_min = Time(jd_trigger + min_days, format='jd')
+    time_max = Time(jd_trigger + max_days, format='jd')
+
+    sources = []
+    sources_data = []
+    for ra, dec in zip(ra_center, dec_center):
+        r = requests.post(
+            urllib.parse.urljoin(cfg['app.fink_endpoint'], 'api/v1/explorer'),
+            json={
+                'ra': ra,
+                'dec': dec,
+                'radius': 60 * radius,
+                'startdate_conesearch': Time(jd_trigger + min_days, format='jd').iso,
+                'window_days_conesearch': max_days,
+            },
+        )
+        objs = pd.DataFrame(r.json())
+        for index, obj in objs.iterrows():
+            objectId = obj['i:objectId']
+            ra_obj, dec_obj = obj['i:ra'], obj['i:dec']
+            jdstarthist = obj['i:jdstarthist']
+            jdendhist = obj['i:jdendhist']
+            if (jdstarthist < time_min.jd) or (jdendhist > time_max.jd):
+                continue
+            if objectId in sources:
+                continue
+            df = query_fink_photometry(objectId)
+            det = np.where(
+                (~np.isnan(df['mag']))
+                & (df['mjd'] >= time_min.mjd)
+                & (df['mjd'] <= time_max.mjd)
+            )[0]
+            ndet = len(det)
+            if ndet >= ndethist_min:
+                sources.append(objectId)
+                sources_data.append(
+                    {'id': objectId, 'ra': ra_obj, 'dec': dec_obj, 'data': df}
+                )
+
+    return sources_data
+
+
+def query_fink_photometry(objectId):
+    """Fetch object photometry from Fink.
+    objectId: str
+        Object ID
+
+    Returns
+    -------
+    df : pandas.DataFrame
+        A dataframe with the object photometry
+    """
+
+    desired_columns = {
+        'i:jd',
+        'i:ra',
+        'i:dec',
+        'i:magpsf',
+        'i:sigmapsf',
+        'i:diffmaglim',
+        'i:magzpsci',
+        'i:fid',
+    }
+
+    r = requests.post(
+        urllib.parse.urljoin(cfg['app.fink_endpoint'], 'api/v1/objects'),
+        json={'objectId': objectId, 'output-format': 'json'},
+    )
+    df = pd.DataFrame.from_dict(r.json())
+
+    if not desired_columns.issubset(set(df.columns)):
+        raise ValueError('Missing expected column')
+
+    df.rename(
+        columns={
+            'i:jd': 'jd',
+            'i:ra': 'ra',
+            'i:dec': 'dec',
+            'i:magpsf': 'mag',
+            'i:sigmapsf': 'magerr',
+            'i:diffmaglim': 'limiting_mag',
+            'i:magzpsci': 'zp',
+            'i:fid': 'filter',
+        },
+        inplace=True,
+    )
+    df['filter'] = [inv_bands[int(filt)] for filt in df['filter']]
+    df['mjd'] = [Time(jd, format='jd').mjd for jd in df['jd']]
+
+    drop_columns = list(
+        set(df.columns.values)
+        - {'mjd', 'ra', 'dec', 'mag', 'magerr', 'limiting_mag', 'filter'}
+    )
+
+    df.drop(
+        columns=drop_columns,
+        inplace=True,
+    )
```

Comment:
Would it be easier to just do 
`df = df[['mjd', 'ra', 'dec', 'mag', 'magerr', 'limiting_mag', 'filter']]`

Reply:
Interesting! I think you had suggested this version for previous ones so we have this form elsewhere. What do you think?

Reply:
I don't remember that, and I don't think there's a difference in runtime or anything like that... 
In this case it looks like you are doing double-negatives and it takes 8 lines instead of one, so the choice in this case is fairly obvious to me.